### PR TITLE
Soft 999 james carr pries fw 102

### PR DESCRIPTION
**Pull request by jcarrpries**

### CODE REVIEW

File: projects/hello_world/src/main.c [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+int main(void) {
+  interrupt_init();
+  soft_timer_init();
+
+  Counters counters;
+  counters.counter_a = 0;
+  counters.counter_b = 0;
+  counters.half_second = false;
+
+  soft_timer_start_millis(500, callback, &counters, NULL);
```
Comment (g-jessmuir) [inline projects/hello_world/src/main.c:40]: please use a `#define ... 500` for the 500 milliseconds with a descriptive name

File: projects/hello_world/src/main.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+#include "soft_timer.h"  // Software timers for scheduling future events.
+#include "wait.h"        // for the wait function
+
+typedef struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+  bool half_second;
+} Counters;
+
+void callback(SoftTimerId timer_id, void *context) {
```
Comment (g-jessmuir) [inline projects/hello_world/src/main.c:13]: please use `static void`, and prefix the function name with `prv_` to denote the function is private to this file/module. Could also be made more descriptive by using `prv_timer_callback`.

File: projects/hello_world/src/main.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
+#include "delay.h"  // For real-time delays
```
Comment (g-jessmuir) [inline projects/hello_world/src/main.c:1]: please remove unnecessary include

File: projects/hello_world/src/main.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  counters->half_second = !counters->half_second;
+
+  soft_timer_start_millis(500, callback, counters, NULL);
+}
+
+int main(void) {
+  interrupt_init();
+  soft_timer_init();
+
+  Counters counters;
```
Comment (g-jessmuir) [inline projects/hello_world/src/main.c:35]: we usually initialize an empty struct with `= { 0 };`, please use this for clarity. For reference, doing this initializes the memory of the struct to zero.

File: projects/hello_world/src/main.c [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  counters.counter_a = 0;
+  counters.counter_b = 0;
+  counters.half_second = false;
```
Comment (g-jessmuir) [inline projects/hello_world/src/main.c:38]: With the above comment, you won't need to explicitly set these values here.

File: projects/hello_world/src/main.c (hunk 4 above)
Comment (jcarrpries) [reply projects/hello_world/src/main.c:35]: will this work for the boolean being set to false? I understand why it would work for just numerical values.

Comment (g-jessmuir) [reply projects/hello_world/src/main.c:35]: yes, a false boolean in C is equivalent to 0

**Comment by g-jessmuir** [comment]
LGTM

**Pull request closed without merging**